packages/api/src/resolvers/filters/index.ts

```diff
@@ -166,6 +176,113 @@ export const filtersResolver = authorized<FiltersSuccess, FiltersError>(
     }
   }
 )
+const updatePosition = async (
+  uid: string,
+  filter: Filter,
+  newPosition: number
+) => {
+  const { position } = filter
+  const moveUp = newPosition < position
+
+  // move filter to the new position
+  const updated = await AppDataSource.transaction(async (t) => {
+    await setClaims(t, uid)
+
+    // update the position of the other filters
+    const updated = await t.getRepository(Filter).update(
+      {
+        user: { id: uid },
+        position: Between(
+          Math.min(newPosition, position),
+          Math.max(newPosition, position)
+        ),
+      },
+      {
+        position: () => `position + ${moveUp ? 1 : -1}`,
+      }
+    )
+
+    if (!updated.affected) {
+      return null
+    }
+
+    // update the position of the filter
+    return t.getRepository(Filter).save({
+      ...filter,
+      position: newPosition,
+    })
+  })
+
+  if (!updated) {
+    throw new Error('unable to update')
+  }
+
+  return updated
+}
+
+export const updateFilterResolver = authorized<
+  UpdateFilterSuccess,
+  UpdateFilterError,
+  MutationUpdateFilterArgs
+>(async (_, { input }, { claims: { uid }, log }) => {
+  const repo = getRepository(Filter)
+  const { id } = input
+
+  try {
+    const user = await getRepository(User).findOneBy({ id: uid })
+    if (!user) {
+      return {
+        errorCodes: [UpdateFilterErrorCode.Unauthorized],
+      }
+    }
+
+    const filter = await getRepository(Filter).findOne({
+      where: { id },
```

Comment:
I think we should have user id here in the where clause too, otherwise users could "guess" uuids and query for other users filters I think.